source/opt/inline_pass.cpp

```diff
@@ -0,0 +1,430 @@
+  ir::Module::inst_iterator type_itr = module_->types_values_begin();
+  for (; type_itr != module_->types_values_end(); ++type_itr) {
+    const ir::Instruction* type_inst = &*type_itr;
+    if (type_inst->opcode() == SpvOpTypePointer &&
+        type_inst->GetOperand(kSpvTypePointerTypeId).words[0] == typeId &&
+        type_inst->GetOperand(kSpvTypePointerStorageClass).words[0] ==
+            SpvStorageClassFunction)
+      break;
+  }
+  return (type_itr != module_->types_values_end()) ? type_itr->result_id() : 0;
+}
+
+uint32_t InlinePass::AddPointerToType(uint32_t typeId) {
+  uint32_t resultId = TakeNextId();
+  std::vector<ir::Operand> in_operands;
+  in_operands.emplace_back(
+      spv_operand_type_t::SPV_OPERAND_TYPE_STORAGE_CLASS,
+      std::initializer_list<uint32_t>{uint32_t(SpvStorageClassFunction)});
+  in_operands.emplace_back(spv_operand_type_t::SPV_OPERAND_TYPE_ID,
+                           std::initializer_list<uint32_t>{uint32_t(typeId)});
+  std::unique_ptr<ir::Instruction> type_inst(
+      new ir::Instruction(SpvOpTypePointer, 0, resultId, in_operands));
+  module_->AddType(std::move(type_inst));
+  return resultId;
+}
+
+void InlinePass::AddBranch(uint32_t labelId,
+                           std::unique_ptr<ir::BasicBlock>* bp) {
+  std::vector<ir::Operand> branch_in_operands;
+  branch_in_operands.push_back(
+      ir::Operand(spv_operand_type_t::SPV_OPERAND_TYPE_ID,
+                  std::initializer_list<uint32_t>{labelId}));
+  std::unique_ptr<ir::Instruction> newBranch(
+      new ir::Instruction(SpvOpBranch, 0, 0, branch_in_operands));
+  (*bp)->AddInstruction(std::move(newBranch));
+}
+
+void InlinePass::AddStore(uint32_t ptrId, uint32_t valId,
+                          std::unique_ptr<ir::BasicBlock>* bp) {
+  std::vector<ir::Operand> store_in_operands;
+  store_in_operands.push_back(
+      ir::Operand(spv_operand_type_t::SPV_OPERAND_TYPE_ID,
+                  std::initializer_list<uint32_t>{ptrId}));
+  store_in_operands.push_back(
+      ir::Operand(spv_operand_type_t::SPV_OPERAND_TYPE_ID,
+                  std::initializer_list<uint32_t>{valId}));
+  std::unique_ptr<ir::Instruction> newStore(
+      new ir::Instruction(SpvOpStore, 0, 0, store_in_operands));
+  (*bp)->AddInstruction(std::move(newStore));
+}
+
+void InlinePass::AddLoad(uint32_t typeId, uint32_t resultId, uint32_t ptrId,
+                         std::unique_ptr<ir::BasicBlock>* bp) {
+  std::vector<ir::Operand> load_in_operands;
+  load_in_operands.push_back(
+      ir::Operand(spv_operand_type_t::SPV_OPERAND_TYPE_ID,
+                  std::initializer_list<uint32_t>{ptrId}));
+  std::unique_ptr<ir::Instruction> newLoad(
+      new ir::Instruction(SpvOpLoad, typeId, resultId, load_in_operands));
+  (*bp)->AddInstruction(std::move(newLoad));
+}
+
+// Generate callee code into newBlocks to be inlined for the function call at
+// call_ii. Also add new function variables into caller func
+
+void InlinePass::GenInlineCode(
```

Comment:
This method is quite large which negatively impacts the readability. Maybe consider break it into several methods?